### PR TITLE
Define BankingTracer::create_channels()

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

unified scheduler wants to treat all the transactions as same. 

#### Summary of Changes

Unify the incoming channels for the banking stage conditionally. Currently stubbed.

No functional change, so no test is added.

extracted from #3946, see the pr for the general overview.

### CODE REVIEW

File: core/src/banking_trace.rs [hunk 1]
```diff
@@ -220,22 +230,85 @@ impl BankingTracer {
...
 
-    pub fn create_channel_tpu_vote(&self) -> (BankingPacketSender, BankingPacketReceiver) {
+    fn create_channel_tpu_vote(&self) -> (BankingPacketSender, BankingPacketReceiver) {
         self.create_channel(ChannelLabel::TpuVote)
     }
 
-    pub fn create_channel_gossip_vote(&self) -> (BankingPacketSender, BankingPacketReceiver) {
+    fn create_channel_gossip_vote(&self) -> (BankingPacketSender, BankingPacketReceiver) {
         self.create_channel(ChannelLabel::GossipVote)
     }
```
Comment (ryoqun) [inline core/src/banking_trace.rs:283]: these fns are now private

**Comment by ryoqun** [comment]
@apfitzge ping :)

File: core/src/banking_trace.rs [hunk 2]
```diff
@@ -220,22 +230,85 @@ impl BankingTracer {
         self.active_tracer.is_some()
     }
 
+    pub fn create_channels(&self, pool: Option<&Arc<DefaultSchedulerPool>>) -> Channels {
```
Comment (apfitzge) [inline core/src/banking_trace.rs]: neat - I actually wanted to experiment with this in central-scheduler as well. Now I don't have to do the work 😄

Comment (apfitzge) [reply core/src/banking_trace.rs]: given that, can we just make this take a bool?
unified-scheduler can have this always be true (Option::is_some) and for central-scheduler can control via some sort of configuration.

**Review by apfitzge — COMMENTED** [review]
See https://github.com/anza-xyz/agave/pull/4041/files#r1882325321

Otherwise, love the change

**Comment by apfitzge** [comment]
> @apfitzge ping :)

ha. was literally typing my comments as you pinged me 😆

File: core/src/banking_trace.rs (hunk 2 above)
Comment (ryoqun) [reply core/src/banking_trace.rs]: done: https://github.com/anza-xyz/agave/pull/4041/commits/1d014047268da11799cbdec23734dcf5f5d78fd2

**Review by apfitzge — APPROVED** [review]

**Pull request merged**